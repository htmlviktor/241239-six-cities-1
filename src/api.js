import axios from 'axios';

export const createAPI = (onLoginFail) => {
  const api = axios.create({
    baseURL: `https://es31-server.appspot.com/six-cities`,
    timeout: 5000,
    withCredentials: true
  });

  const onSuccess = (responce) => responce;
  const onFail = (err) => {
    if (err.response.request.responseURL.indexOf(`/login`) === -1 && err.response.status === 403) {
      onLoginFail();
      return;
    }
    throw err;
  };
  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
