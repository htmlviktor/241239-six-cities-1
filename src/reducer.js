import {loadOffers} from './actions';

const initialState = {
  currentCity: `Amsterdam`,
  listCities: new Set([...data.map((offer) => offer.city)]),
  offers: []
};

const Operation = {
  loadOffers: () => (dispatch, _getState, api) => {
    return api.get(`/hotels`)
      .then((response) => {
        dispatch(loadOffers(response.data));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `CITY_CHANGE`:
      return Object.assign({}, state, {
        currentCity: action.city
      });
    case `LOAD_OFFERS`:
      return Object.assign({}, state, {
        offers: action.offers
      });
    default: return state;
  }
};


export {
  reducer,
  Operation
};
