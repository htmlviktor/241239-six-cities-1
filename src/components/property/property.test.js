import React from 'react';
import renderer from 'react-test-renderer';
import Property from './property.jsx';

import {Router} from 'react-router-dom';
import history from '../../history';

import {createStore, applyMiddleware} from 'redux';
import {Provider} from 'react-redux';
import reducer from '../../reducer/index';
import {createAPI} from '../../api';
import thunk from 'redux-thunk';

const mockCities = new Set([
  `Amsterdam`,
  `Budapesht`,
  `Wroclaw`
]);

const mockOffers = [
  {
    isPremium: true,
    previewImage: ``,
    price: 222,
    isFavorite: true,
    id: 1,
    title: ``,
    type: ``,
    city: {
      name: `Wroclaw`
    }
  },
  {
    isPremium: true,
    previewImage: ``,
    price: 222,
    isFavorite: true,
    id: 2,
    title: ``,
    type: ``,
    city: {
      name: `Wroclaw`
    }
  },
];

const api = createAPI();
const store = createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));


it(`Test Property Component`, () => {
  const tree = renderer
  .create(
      <Provider store={store}>
        <Router history={history}>
          <Property cities={mockCities} offers={mockOffers}/>
        </Router>
      </Provider>
  )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
