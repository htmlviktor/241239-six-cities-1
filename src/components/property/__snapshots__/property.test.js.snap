// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Property Component 1`] = `null`;
